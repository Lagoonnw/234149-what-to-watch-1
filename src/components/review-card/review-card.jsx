import React        from 'react';
import {formatDate} from '../../helpers/format-date/format-date';

export const ReviewCard = ({comment, user, rating, date}) => (
  <div className="review">
    <blockquote className="review__quote">
      <p className="review__text">{comment}</p>

      <footer className="review__details">
        <cite className="review__author">{user.name}</cite>
        <time
          className="review__date"
          dateTime={`${formatDate(date).localFormat}`}
        >
          {`${formatDate(date).comprehensible}`}
        </time>
      </footer>
    </blockquote>

    <div className="review__rating">{`${rating.toFixed(1)}`}</div>
  </div>
);

