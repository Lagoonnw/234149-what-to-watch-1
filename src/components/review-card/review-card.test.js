import React from 'react';
import renderer from 'react-test-renderer';
import {ReviewCard} from './review-card.jsx';

test(`Should render Review card correctly`, () => {
  const tree = renderer.create(
      <ReviewCard
        user={{id: 1, name: `Sam`}}
        rating={8.75454}
        comment={`Comment`}
        date={`2019-06-11T15:12:25.115Z`}
      />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
