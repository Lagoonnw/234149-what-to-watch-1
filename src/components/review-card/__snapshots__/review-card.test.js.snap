// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Review card correctly 1`] = `
<div
  className="review"
>
  <blockquote
    className="review__quote"
  >
    <p
      className="review__text"
    >
      Comment
    </p>
    <footer
      className="review__details"
    >
      <cite
        className="review__author"
      >
        Sam
      </cite>
      <time
        className="review__date"
        dateTime="6/11/2019"
      >
        June 11, 2019
      </time>
    </footer>
  </blockquote>
  <div
    className="review__rating"
  >
    8.8
  </div>
</div>
`;
